### PR TITLE
Add pine protocol support

### PR DESCRIPTION
**Pull request by jupiteruwdev**
**NOTE**

> If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).

1. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
2. The protocol is usually listed within 24 hours of merging the PR
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Logo (High resolution, preferably in .svg and .png, for application on both white and black backgrounds. Will be shown with rounded borders):
Logo-Horizontal-Colour

##### Short Description (to be shown on DefiLlama):
Non-custodial decentralized asset-backed lending protocol that allows borrowers to borrow fungible digital tokens from lenders using non-fungible tokens as collateral

##### Category (full list at https://defillama.com/categories) *Please choose only one:
NFT Lending

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/pine-protocol exports TVL: 
        
 
 
    ethereum                  496.05 k
    
    total                    496.05 k

**Comment by g1nt0ki** [comment]
@jupiteruwdev can you please fill in  the project details in PR description

**Comment by g1nt0ki** [comment]
how are you pricing the NFTs? can you update the adapter to compute tvl using onchain calls, let me know if you need some help wth that

**Review by g1nt0ki — CHANGES_REQUESTED** [review]
.

**Comment by jupiteruwdev** [comment]
> 

We've a lot of pools and we don't have an aggregated pool or contract for calculating tvl. But we're calculating in our backend service which is currently used in tvl function.

**Comment by jupiteruwdev** [comment]
> how are you pricing the NFTs? can you update the adapter to compute tvl using onchain calls, let me know if you need some help wth that

We've a lot of pools and we don't have an aggregated pool or contract for calculating tvl. But we're calculating in our backend service which is currently used in tvl function.

**Comment by g1nt0ki** [comment]
right now it is a complete blackbox, can you expose the pool list via an api? and then use it in the adapter? Also, how are you pricing the NFTs?

**Comment by jupiteruwdev** [comment]
> right now it is a complete blackbox, can you expose the pool list via an api? and then use it in the adapter? Also, how are you pricing the NFTs?

Sure, will do. We're relying on third party resources like - alchemy, opensea, gemxyz, ...

**Comment by llamatester** [comment]
Error while running adapter at projects/pine-protocol: 
        
 
  
    
    
    Error: connect ECONNREFUSED 127.0.0.1:8080
    Truncated error stack:
    
        at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16)

**Review by g1nt0ki — CHANGES_REQUESTED** [review]

File: projects/pine-protocol/index.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  }
+`;
+
+const provider = new ethers.providers.JsonRpcProvider(
+  "https://eth-mainnet.g.alchemy.com/v2/UN5z0qze-a7hJuNOqFMUs9t2blMsxcN7"
+);
+
+const tvl = async () => {
+  const { data: pools } = await axios.get(
+    "http://localhost:8080/v0/pools/addresses?ethereum=1"
```
[the remark this answers is not in the document]
Comment (jupiteruwdev) [reply projects/pine-protocol/index.js:33]: Yes, the api is not deployed on prod so I'll update it later whenever it's deployed.

**Comment by g1nt0ki** [comment]
@jupiteruwdev I have updated it to use logs here: https://github.com/DefiLlama/DefiLlama-Adapters/commit/ec8f0a6a5ed48627ebb6ba1533ded4ffecf111ae

**Pull request merged**